src/Microsoft.Azure.SignalR.Emulator/Program.cs

```diff
@@ -1,35 +1,192 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.IO;
-
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.CommandLineUtils;
 using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.SignalR.Emulator
 {
     public class Program
     {
+        private const int DefaultPort = 8888;
+        private static readonly string SettingsFileName = "settings.json";
+        private static readonly string SettingsFile = Path.GetFullPath(SettingsFileName);
+        private static readonly string ProgramDefaultSettingsFile = Path.Combine(AppContext.BaseDirectory, SettingsFileName);
+        private static readonly string AppSettingsFile = Path.Combine(AppContext.BaseDirectory, "appsettings.json");
+
         public static void Main(string[] args)
         {
             // todo: "upstream init" to generate the default setting.json file
             // todo: "upstream list" to list the settings file
             // todo: "start" to run the emulator
             // todo: "help" to explain the upstream and pattern rules
-            CreateHostBuilder(args).Build().Run();
+
+            var app = new CommandLineApplication();
+            app.Name = "asrs-emulator";
+            app.Description = "The local emulator for Azure SignalR Serverless features.";
+            app.HelpOption("-h|--help");
+            
+            app.Command("upstream", command =>
+            {
+                command.Description = "To init/list the upstream options";
+                command.HelpOption("-h|--help");
+                command.Command("init", c =>
+                {
+                    c.Description = "Init the default upstream options into a settings.json config. Use -o to specify the folder to export the default settings.";
+                    var configOptions = c.Option("-o|--output", "Specify the folder to init the upstream settings file.", CommandOptionType.SingleValue);
+                    c.HelpOption("-h|--help");
+                    c.OnExecute(() =>
+                    {
+                        string outputFile = configOptions.HasValue() ? Path.GetFullPath(Path.Combine(configOptions.Value(), SettingsFileName)) : SettingsFile;
+                        if (File.Exists(outputFile))
+                        {
+                            Console.WriteLine($"Already contains '{outputFile}', still want to override it with the default one? (N/y)");
```

Comment:
Why the `N` is upper case while `y` is lower case

Reply:
upper case is the default one